vk/api.py

```diff
@@ -2,155 +2,45 @@
-
-                elif error.is_access_token_incorrect():
-                    logger.info('Authorization failed. Access token will be dropped')
-                    self.access_token = None
-                    return self.make_request(method_request)
-
-                else:
-                    raise error
-
-    def send_api_request(self, request, captcha_response=None):
-        url = self.API_URL + request._method_name
-        method_args = request._api._method_default_args.copy()
-        method_args.update(stringify_values(request._method_args))
-        access_token = self.access_token
-        if access_token:
-            method_args['access_token'] = access_token
-        if captcha_response:
-            method_args['captcha_sid'] = captcha_response['sid']
-            method_args['captcha_key'] = captcha_response['key']
-        timeout = request._api._timeout
-        response = self.requests_session.post(url, method_args, timeout=timeout)
-        return response
-
-    def get_captcha_key(self, captcha_image_url):
-        """
-        Default behavior on CAPTCHA is to raise exception
-        Reload this in child
-        """
-        return None
-    
-    def auth_code_is_needed(self, content, session):
-        """
-        Default behavior on 2-AUTH CODE is to raise exception
-        Reload this in child
-        """           
-        raise VkAuthError('Authorization error (2-factor code is needed)')
-    
-    def auth_captcha_is_needed(self, content, session):
-        """
-        Default behavior on CAPTCHA is to raise exception
-        Reload this in child
-        """              
-        raise VkAuthError('Authorization error (captcha)')
-    
-    def phone_number_is_needed(self, content, session):
-        """
-        Default behavior on PHONE NUMBER is to raise exception
-        Reload this in child
-        """
-        logger.error('Authorization error (phone number is needed)')
-        raise VkAuthError('Authorization error (phone number is needed)')
-
-
 class API(object):
     def __init__(self, session, timeout=10, **method_default_args):
         self._session = session
         self._timeout = timeout
         self._method_default_args = method_default_args
 
+    @classmethod
+    def create_api(cls, app_id=None, login=None, password=None, timeout=10,
+                   **method_default_args):
+        """Factory method to explicitly create API with app_id, login and
+        password parameters.
+        If those are not passed - token-free session will be created
+        automatically
+
+        :return: vk.API instance
+        """
+        session = VKSession(app_id, login, password)
+        instance = cls(session=session, timeout=timeout, **method_default_args)
+        return instance
+
+    def get_default_args(self):
+        return self._method_default_args.copy()
+
+    def get_timeout(self):
```

Comment:
pardon, timeout of what ?
Btw, this is a good point, the `API.__call__` can be safely removed at all, cuz nothing uses it, all request calls are based on `Request.__call__`